### ISSUE TITLE
Alternative on-site Hubbard U input not passed to SIRIUS

### ISSUE THREAD

**Issue by gsavva** [electronic-structure/q-e-sirius#33]
In Quantum ESPRESSO, the following alternatives for defining on-site Hubbard U are equivalent:
1. `U Co-3d 5.0`
2. `V Co-3d Co-3d 1 1 5.0`

However, only the 1st one is passed correctly to SIRIUS.
With the 2nd syntax, the Hubbard correction is enabled, but the value of U is not correct:

SIRIUS output using **syntax 1**
```
[atom type] Hubbard correction               : true
[atom type]   angular momentum                   : 2
[atom type]   principal quantum number           : 3
[atom type]   occupancy                          : 7
[atom type]   U                                  : 0.183747         <===============
[atom type]   number of hubbard radial functions : 1
[atom type]   number of hubbard basis functions  : 5
[atom type]   Hubbard wave-functions             : {n: 3, l: 2}
[atom type]   orthogonalize                      : true
[atom type]   normalize                          : false
[atom type]   full_orthogonalization             : true
[atom type]   simplified                         : true
```
```
!    total energy              =    -373.17517843 Ry
...
     Hubbard energy            =       0.18143765 Ry
```
SIRIUS output using **syntax 2**
```
[atom type] Hubbard correction               : true
[atom type]   angular momentum                   : 2
[atom type]   principal quantum number           : 3
[atom type]   occupancy                          : 7
[atom type]   U                                  : 0                <===============
[atom type]   number of hubbard radial functions : 1
[atom type]   number of hubbard basis functions  : 5
[atom type]   Hubbard wave-functions             : {n: 3, l: 2}
[atom type]   orthogonalize                      : true
[atom type]   normalize                          : false
[atom type]   full_orthogonalization             : true
[atom type]   simplified                         : false
```
```
!    total energy              =    -373.36903877 Ry

     Hubbard energy            =      -0.00000000 Ry
```
Inputs, outputs and pseudos attached: 
[LiCoO2_in-out-pseudos.zip](https://github.com/electronic-structure/q-e-sirius/files/12334374/LiCoO2_in-out-pseudos.zip)

**Comment by mtaillefumier** [comment]
the second syntax is misleading. If you want to have onsite hubbard, then use the first notation which is clearer than the second one. So as a workaround for the time being (I will modify the sirius interface) or find out why the intersite hubbard gives zero use the first notation for the onsite and the second notation for the V term. Do not use the second notation for the onsite one.

NB : I can understand why you may want the second notation to work with onsite (for instance the same atom with a site dependent U) but then create two atom types instead of using this trick. The input file would certainly gain in clarity

**Comment by gsavva** [comment]
@mtaillefumier thanks for modifying the q-e-sirius interface in #34.

I came across the above behavior while running scf+Hubbard benchmarks via `aiida`. The tool that creates the Hubbard card explicitly requires on-site U interactions to be defined using the 2nd syntax above in the presence of inter-site (+V) interactions. ([reference](https://github.com/aiidateam/aiida-quantumespresso/blob/7263f92ccabcfc9f828b9da5473e1aefbc4b8eca/src/aiida_quantumespresso/utils/hubbard.py#L69-L72)).

In fact, the LiCoO2 example I used above ([example01](https://gitlab.com/QEF/q-e/-/tree/qe-7.2/HP/examples/example01/reference) from QE repo) has only a U value, so in similar cases where only on-site U interactions are included, `aiida` would create an input file using the syntax `"U atom-orbital U-value"`.

[Example10](https://gitlab.com/QEF/q-e/-/tree/qe-7.2/HP/examples/example10/reference), though, includes on-site U and inter-site V and it is, thus, a more appropriate and representative case.

Modifying the HUBBARD card of the input of example10 as follows:

**syntax_1_U**
```
HUBBARD {ortho-atomic}
U  Co-3d               7.7514
V  Co-3d   O-2p  1  2  0.7573
```
**syntax_2_V**
```
HUBBARD {ortho-atomic}
V  Co-3d  Co-3d  1  1  7.7514
V  Co-3d   O-2p  1  2  0.7573
```
I run native QE 7.2 and q-e-sirius after #34.

- Native QE 7.2 gives exactly the same results.
- q-e-sirius results differ substantially, namely, the `Hubbard energy` is almost double when using **syntax_1_U** than **syntax_2_V** as shown below:

**q-e-sirius, syntax_1_U:**
``` 
!    total energy              =    -372.82055187 Ry
     estimated scf accuracy    <          0.0E+00 Ry

     The total energy is the sum of the following terms:
     one-electron contribution =    -252.11122871 Ry
     hartree contribution      =     141.23777933 Ry
     xc contribution           =     -51.37127318 Ry
     ewald contribution        =    -211.06189316 Ry
     Hubbard energy            =       0.48606341 Ry      <===============
     scf correction            =       0.00000046 Ry
```
**q-e-sirius, syntax_2_V**
```
!    total energy              =    -373.07703571 Ry
     estimated scf accuracy    <          0.0E+00 Ry

     The total energy is the sum of the following terms:
     one-electron contribution =    -253.03807379 Ry
     hartree contribution      =     142.14684535 Ry
     xc contribution           =     -51.39413705 Ry
     ewald contribution        =    -211.06189316 Ry
     Hubbard energy            =       0.27021910 Ry      <===============
     scf correction            =       0.00000385 Ry
```

All files used to get the above results are attached here
[ex10_files.zip](https://github.com/electronic-structure/q-e-sirius/files/12407823/ex10_files.zip)

Actually the above may be a separate issue, unrelated to how input is parsed (?). What do you think @mtaillefumier ?

**Comment by mtaillefumier** [comment]
the only way to solve this is to fix the aiida plugin to write a proper config file. SIRIUS makes a clear distinction between the onsite and intersite interaction as it should be. The reason for this is simple onsite interaction imposes the initialization of the occupation matrix to some predefined values. I had to hack qe to give the right information to sirius so the issue is *NOT* with sirius but QE and aiida.

**Comment by mtaillefumier** [comment]
```
# This variable is to meet QE implementation. If intersite interactions
            # (+V) are present, onsite parameters might not be relabelled by the ``hp.x``
            # code, causing a subsequent ``pw.x`` calculation to crash. That is,
            # we need to avoid writing "U Co-3d 5.0", but instead "V Co-3d Co-3d 1 1 5.0".
```
this comment is revealing. Fixing QE to behave normally is the right way to deal with the problem not asking devs to implement misbehaviors in their code.

**Comment by gsavva** [comment]
Regarding the `Hubbard energy` difference reported above, is this likely to be linked with how information about U and V is passed to SIRIUS ?

**Comment by mtaillefumier** [comment]
most likely.

**Comment by mtaillefumier** [comment]
should be fixed with #35. This bug in QE should really be fixed.

**Comment by toxa81** [comment]
Fixed
